File: backend/LogisticsAssetTracker.Api/Domain/Entities/Location.cs
using LogisticsAssetTracker.Api.Domain.Enums;

namespace LogisticsAssetTracker.Api.Domain.Entities;

public class Location
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public LocationType Type { get; set; }
    public string? Address { get; set; }
    public string? Description { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: frontend/src/types/location.ts
export type LocationType =
  | "Warehouse"
  | "Truck"
  | "DistributionCenter"
  | "ClientSite"
  | "MaintenanceArea"
  | "Unknown";

export interface Location {
  id: string;
  name: string;
  type: LocationType;
  address?: string | null;
  description?: string | null;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateLocationRequest {
  name: string;
  type: LocationType;
  address?: string;
  description?: string;
}

export interface UpdateLocationRequest {
  name: string;
  type: LocationType;
  address?: string;
  description?: string;
}

export interface LocationListQuery {
  type?: LocationType;
  isActive?: boolean;
  search?: string;
  page?: number;
  pageSize?: number;
  sortBy?: string;
  sortDirection?: "asc" | "desc";
}
